lib/collections/teams.js

```diff
@@ -636,6 +636,67 @@ Meteor.methods({
     });
   },
 
+  // compute the score for a hypothetical puzzle response from a
+  // team. Returns the computed score. This Method does not update any
+  // database state
+  'admin.team.computeScore'(puzzle) {
+    check(puzzle, Object);
+    requireAdmin()
+    if (!Meteor.isServer) {
+      return true;
+    }
+    
+    const { puzzleId } = puzzle;
+    const masterPuzzle = Puzzles.findOne(puzzleId);
+    const endTime = puzzle.end;
+    const [tempTimeout, tempScore] = getPuzzleScoreAndTimeout(puzzle, endTime, masterPuzzle, false);
+    return [tempTimeout, tempScore];
+  },
+
+  // update the start time, end time, and hints taken information for
+  // one team and puzzle.
+  'admin.team.updatePuzzlePlay'(teamId, puzzle) {
+    check(teamId, String);
+    check(puzzle, Object);
+    requireAdmin();
+    if (!Meteor.isServer) {
+      return true;
+    }
+
+    const team = Teams.findOne(teamId);
+    if (!team) {
+      throw new Meteor.Error(400, `Oops, no team found with id ${teamId}`);
+    }
+    const { puzzleId } = puzzle;
+    const i = findIndex(team.puzzles, (p) => p.puzzleId === puzzleId);
+    if (!i) {
+      throw new Meteor.Error(400, `Oops, no puzzle found with id ${puzzleId}`);
+    }
+
+    const puzzleName = puzzle.name;
+    const newStart = puzzle.start;
+    const newEnd = puzzle.end;
+    const newHints = puzzle.hints;
+    const masterPuzzle = Puzzles.findOne(puzzleId);
+    const [newTimeout, newScore] = getPuzzleScoreAndTimeout(puzzle, newEnd, masterPuzzle, false);
+
+    let s='';
+    newHints.forEach((hint, index) => {
+      s += (hint.taken ? '1' : '0');
+    });
+    Meteor.logger.info(`updating puzzle play info for team ${teamId} puzzle ${puzzleName}, start ${newStart}, end ${newEnd}, hints ${s}`);
```

Comment:
There is a `teamShortName` function or similar that makes the logs easier to read